simphony_ui/ui.py

```diff
@@ -0,0 +1,199 @@
+from traits.api import (HasStrictTraits, Int, Float, Enum, Instance,
+                        Str, Directory, BaseInt, BaseFloat, Array, File)
+from traitsui.api import View, Item, UItem, Tabbed, VGroup, HGroup
+import os
+
+
+class PositiveFloat(BaseFloat):
+    """ A custom trait representing a positive float """
+
+    default_value = 1.0
+
+    info_text = 'a positive float'
+
+    def validate(self, object, name, value):
+        value = super(PositiveFloat, self).validate(object, name, value)
+        if value > 0.0:
+            return value
+
+        self.error(object, name, value)
+
+
+class PositiveInt(BaseInt):
+    """ A custom trait representing a positive integer """
+
+    default_value = 1
+
+    info_text = 'a positive integer'
+
+    def validate(self, object, name, value):
+        value = super(PositiveInt, self).validate(object, name, value)
+        if value > 0:
+            return value
+
+        self.error(object, name, value)
+
+
+class GlobalParametersModel(HasStrictTraits):
+    """ The model of Global input parameters """
+
+    num_iterations = Int(10)
+
+    force_type = Enum('Stokes', 'Coul', 'Dala')
+    force_type.default_value = 'Stokes'
+
+    traits_view = View(
+        VGroup(
+            Item(name='num_iterations', label='Number of iterations'),
+            Item(name='force_type'),
+        )
+    )
+
+
+class LiggghtsModel(HasStrictTraits):
+    """ The model of Liggghts input parameters """
+
+    timestep = Float(1e-6)
+    num_iterations = Int(10)
+
+    input_file = File()
+
+    # TODO: Display min value textbox when shrink-wrapped option is used
+    boundary_condition_x = Enum('periodic', 'fixed', 'shrink-wrapped')
+    boundary_condition_x.default_value = 'periodic'
+    boundary_condition_y = Enum('periodic', 'fixed', 'shrink-wrapped')
+    boundary_condition_y.default_value = 'fixed'
+    boundary_condition_z = Enum('periodic', 'fixed', 'shrink-wrapped')
+    boundary_condition_z.default_value = 'periodic'
+
+    # TODO: The label is not displayed properly: fix it
+    boundary_condition = HGroup(
```

Comment:
you can technically do this (isolate part of the view in a variable, but in this case I would just create the view once and for all.